src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -0,0 +1,1103 @@
+    ----------
+    data : array-like (1-dimensional)
+    dtype : str, numpy.dtype, or ExtensionDtype, optional
+        Data type for the output Index. If not specified, this will be
+        inferred from `data`.
+        See the :ref:`user guide <basics.dtypes>` for more usages.
+    copy : bool, default False
+        Copy input data.
+    name : object
+        Name to be stored in the index.
+    tupleize_cols : bool (default: True)
+        When True, attempt to create a MultiIndex if possible.
+
+    Notes
+    -----
+    An Index instance can **only** contain hashable objects.
+    An Index instance *can not* hold numpy float16 dtype.
+
+    Examples
+    --------
+    >>> pd.Index([1, 2, 3])
+    Index([1, 2, 3], dtype='int64')
+
+    >>> pd.Index(list('abc'))
+    Index(['a', 'b', 'c'], dtype='object')
+
+    >>> pd.Index([1, 2, 3], dtype="uint8")
+    Index([1, 2, 3], dtype='uint8')
+    """
+
+    # same fields as native pandas index constructor
+    def __init__(
+        self,
+        # Any should be replaced with SnowflakeQueryCompiler when possible (linter won't allow it now)
+        data: ArrayLike | Any = None,
+        dtype: str | np.dtype | ExtensionDtype | None = None,
+        copy: bool = False,
+        name: object = None,
+        tupleize_cols: bool = True,
+    ) -> None:
+        from snowflake.snowpark.modin.plugin.compiler.snowflake_query_compiler import (
+            SnowflakeQueryCompiler,
+        )
+
+        # TODO: SNOW-1359041: Switch to lazy index implementation
+        if isinstance(data, native_pd.Index):
+            self._index = data
+        elif isinstance(data, Index):
+            self._index = data.to_pandas()
+        elif isinstance(data, SnowflakeQueryCompiler):
+            self._index = data._modin_frame.index_columns_index
+        else:
+            self._index = native_pd.Index(
+                data=data,
+                dtype=dtype,
+                copy=copy,
+                name=name,
+                tupleize_cols=tupleize_cols,
+            )
+
+    def to_pandas(self) -> native_pd.Index:
+        """
+        Convert Snowpark pandas Index to pandas Index
+
+        Returns:
+            pandas Index
+        """
+        return self._index
+
+    @property
+    def values(self) -> ArrayLike:
+        """
+        Return an array representing the data in the Index.
+
+        Returns
+        -------
+        array: numpy.ndarray or ExtensionArray
+
+        See Also
+        --------
+        Index.array : Reference to the underlying data.
+
+        Examples
+        --------
+        For :class:`pd.Index`:
+
+        >>> idx = pd.Index([1, 2, 3])
+        >>> idx
+        Index([1, 2, 3], dtype='int64')
+        >>> idx.values
+        array([1, 2, 3])
+        """
+        return self.to_pandas().values
+
+    @property
+    def is_unique(self) -> bool:
+        """
+        Return if the index has unique values.
+
+        Returns
+        -------
+        bool
```

Comment:
make sure your documentation styles match in all the functions - some of them have a description in them that describes what it being returned, others only have a return type.